imod/mf6/boundary_condition.py

```diff
@@ -64,6 +67,15 @@ class BoundaryCondition(Package, abc.ABC):
     not the array input which is used in :class:`Package`.
     """
 
+    def __init__(self, allargs=None):
+        super().__init__(allargs)
+        if "concentration" in allargs.keys() and allargs["concentration"] is None:
+            # Remove vars inplace
+            del self.dataset["concentration"]
+            del self.dataset["concentration_boundary_type"]
+        else:
```

Comment:
if concentration is not present in allargs.keys(), should we do the expand_transient_auxiliary_variables?

Reply:
but now expand_transient_auxiliary_variables(self) is in the else branch. It will do it if concentration is not present in allargs.keys()